File: backend/crawl/app/models/lineup.py
from sqlalchemy import Column, Integer, String, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.models.database import Base


class Lineup(Base):
    __tablename__ = "lineup"

    id = Column(Integer, primary_key=True, index=True)
    game_id = Column(Integer, ForeignKey("game.id"))
    team_id = Column(Integer, ForeignKey("team.id"))
    player_id = Column(Integer, ForeignKey("player.id"))
    batting_order = Column(Integer)
    position = Column(String(10))
    war = Column(String(10))
    
    game = relationship("Game", back_populates="lineup")
    team = relationship("Team", foreign_keys=[team_id])
    player = relationship("Player", foreign_keys=[player_id])

File: backend/baseball/src/main/java/com/newbie/baseball/domain/lineup/entity/LineUp.java
package com.newbie.baseball.domain.lineup.entity;

import com.newbie.baseball.domain.game.entity.Game;
import com.newbie.baseball.domain.player.entity.Player;
import com.newbie.baseball.domain.team.entity.Team;
import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "lineup")
public class LineUp {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "game_id")
    private Game game;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "player_id")
    private Player player;

    @Column(name = "batting_order")
    private Integer battingOrder;

    @Column(name = "position", length = 10)
    private String position;

    @Column(name = "war", length = 10)
    private String war;
}
